malevich/interpreter/local.py

```diff
@@ -0,0 +1,144 @@
+    step: bool = False,
+    *args,
+) -> None:
+    cout(
+        _actions[action],
+        message,
+        verbosity=VerbosityLevel.AllSteps if step else VerbosityLevel.OnlyStatus,
+        level=_levels[level],
+        *args,
+    )
+
+registry = Registry()
+
+class LocalInterpreter(CoreInterpreter):
+    def __init__(self) -> None:
+        super(CoreInterpreter, self).__init__(LocalInterpreterState())
+        self._state = LocalInterpreterState()
+        self.update_state()
+
+    def before_interpret(self, state: LocalInterpreterState) -> LocalInterpreterState:
+        _log("Interpretation in local mode", 0, 0, True)
+        return state
+
+    def create_node(
+        self,
+        state: LocalInterpreterState,
+        tracer: traced[BaseNode]
+    ) -> LocalInterpreterState:
+
+        match tracer.owner:
+            case OperationNode():
+                if tracer.owner.alias is None:
+                    tracer.owner.alias = unique(tracer.owner.processor_id)
+
+                extra = registry.get(tracer.owner.operation_id)
+                if extra is not None:
+                    if 'package_path' not in extra:
+                        raise ValueError(
+                            f'Processor {tracer.owner.processor_id} has '
+                            'installed with installer different from "local". '
+                            'Using local mode is only available for local '
+                            'installations.'
+                        )
+
+                    package_path = Path(extra['package_path'])
+
+                    if not package_path.exists():
+                        raise FileNotFoundError(
+                            f'Processor {tracer.owner.processor_id} has '
+                            f'been installed from path {package_path}, which '
+                            'does not exist.'
+                        )
+
+                else:
+                    raise ValueError(
+                        f'Failed to find processor {tracer.owner.processor_id} '
+                        'in the registry.'
+                    )
+
+                state.import_paths.add(str(package_path.absolute()))
+
+                if tracer.owner.is_condition:
+                    state.conditions[tracer.owner.alias] = Condition(
+                        arguments={},
+                        cfg=json.dumps(tracer.owner.config),
+                        image=JsonImage(ref='', user='', token=''),
+                        conditionId=tracer.owner.processor_id
+                    )
+                else:
+                    state.processors[tracer.owner.alias] = Processor(
+                        arguments={},
+                        cfg=json.dumps(tracer.owner.config),
+                        image=JsonImage(ref=''),
+                        processorId=tracer.owner.processor_id
+                    )
+
+                state.results[tracer.owner.alias] = [Result(name=tracer.owner.alias)]
```

Comment:
for conditions it doesn't make sense - can be only for processors (similarly in the core interpreter)